packages/api-server/api_server/models/door_state.py

```diff
@@ -4,21 +4,22 @@
 from tortoise.models import Model
 
 from ..ros_time import py_to_ros_time, ros_to_py_datetime
-from .door_mode import DoorModeEnum
 
 
 class DoorState(Model):
-    # TODO: foreign key with Door
-    door_name = fields.CharField(255)
-    current_mode = fields.IntEnumField(DoorModeEnum)
+    # KP: We could have a FK to Door,
+    # but I decided against it to make things easier (pesky race conditions).
+    # If we receive a door name that does not exist in the building,
+    # just take it at face value and write into the database regardless.
```

Comment:
do you mean to make `door_name` a nullable FK? A door state will always contains a door_name though, but the problem is that the door_name may not be registered yet because rmf does not have a enrollment phase. So there is a race condition if we receive the door state before the updated building map.

Reply:
I meant keeping the `door_name` but adding another prop that is called `door_fk` to improve the query performance in the future. But I get now and yeah, you're right.